OnlineSchoolPogorzhelskyy/src/hw15/models/Repo.java

```
@@ -0,0 +1,94 @@
+package hw15.models;
+
+
+import javax.swing.text.html.parser.Entity;
+import java.util.Arrays;
+
+public class Repo<E extends Model> implements BaseStorage<E> {
+    private E[] entity;
+
+    public Repo(E[] entity) {
+        this.entity = entity;
+    }
+
+    public int size() {
+        return entity.length;
+    }
+
+    public boolean isEmpty() {
+        for (E e : entity) {
+            if (e != null) return false;
+        }
+        return true;
+    }
+
+    public void add(E instance) {
+        //     if (E.getCounter() == (entity.length - 1))
+        //         entity = Arrays.copyOf(entity, (entity.length * 3 / 2) + 1);
+        //     entity[E.getCounter()-1] = instance;
+        int i = 0;
+        while (i < entity.length) {
+            if (entity[i] == null) {
+                entity[i] = instance;
+                return;
+            }
+            i++;
+        }
+        entity = Arrays.copyOf(entity, (entity.length * 3 / 2) + 1);
+        entity[i] = instance;
+    }
+
+    public void add(int index, E instance) {
+        entity[index] = instance;
+    }
+
+
+
+    public void list() {
+        for (E i : entity) {
+            if (i != null) System.out.println(i);
+        }
+    }
+
+    public E[] getAll() {
+        return entity;
+    }
+
+    public E get(int index) {
+        return entity[index];
+    }
+
+    public E getById(int id)  {
+        for (E i : entity) {
+            if (i != null && i.getId() == id) return i;
+        }
+        //return null;
+        try {
+            throw new EntityNotFoundExeption(id);
```

null should not be returned if there is an error
It's better to just throw the error and process it to the class where the method is called